lib/pipe.js

```diff
@@ -0,0 +1,38 @@
+import {createWriteStream} from 'node:fs';
+import {ChildProcess} from 'node:child_process';
+import {isWritableStream} from 'is-stream';
+
+const isExecaChildProcess = target => target instanceof ChildProcess && isWritableStream(target.stdin) && typeof target.then === 'function';
```

Comment:
Not sure whether this is the best way to check this, considering `instanceof` is not cross-realm safe. However, checking individual properties for existence seems brittle. We could also either use a `Symbol` key on Execa child processes, or a global `WeakSet`.

Reply:
`Symbol` also does not work cross-realm.

Reply:
I don't think cross-realm is a big concern. A lot of stuff don't work cross-realm.

Reply:
Yes, right, neither a `Symbol` nor a global `WeakSet` would work cross-realm! :man_facepalming: 
The only cross-realm solution would be to detect for properties like `pid` or `stdio`.

In this case, the only situation where users would pass child processes cross-realm would be through `vm`, which seems like an unlikely edge case, so maybe we can ignore it.